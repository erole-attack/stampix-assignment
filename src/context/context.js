import React, { useState } from 'react'

export const LanguageContext = React.createContext()

export const LanguageContextProvider = ({ children }) => {
   const [language, setLanguage] = useState('EN');
   return (
      <LanguageContext.Provider value={[language, setLanguage]}>{children}</LanguageContext.Provider>
   )
}