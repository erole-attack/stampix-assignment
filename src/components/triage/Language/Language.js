import React, { useContext, useState } from 'react'
import { primary, secondary } from '../../../styles/colors'

import { LanguageContext } from '../../../context/context'
import { device } from '../../../styles/device'
import styled from 'styled-components'

const Selector = styled.div`
  justify-self: end;
`

const LanguageButtonWrapper = styled.button`
  border: none;
  background: none;
  color: ${props => props.selected ? secondary.sunny_yellow : primary.white};
  font-weight: ${props => props.selected && 700};
  font-size: 1vw;
  margin: 0 0.3vw 0 0.3vw;

  &:hover {
    cursor: pointer;
    color: ${secondary.blushing_pink}
  }
  
  @media ${device.tablet} { 
    font-size: 1.2vw;
    margin: 0 0.1vw 0 0.1vw;
  }
  @media ${device.mobile} { 
    font-size: 12px;
  }
`

const LanguageButton = (props) => {

  const [language] = useContext(LanguageContext);

  return(
    <LanguageButtonWrapper selected={language === props.value ? true : false} onClick={() => props.onClick()}>
      {props.value}
    </LanguageButtonWrapper>
  )
}

const LanguageSwapper = (props) => {

  const [language, setLanguage] = useContext(LanguageContext);

  const clickHandler = (selection) => {
    setLanguage(selection)
  }

  return(
    <>
      {props.options.map(option => 
        <LanguageButton 
          key={option}
          value={option}
          onClick={() => clickHandler(option)}
        />
      )}
    </>
  )
}

export default function Language() {

   return (
        <Selector>
            <LanguageSwapper options={['NL', 'FR', 'EN']}/>
        </Selector>
   )
 }

 export { LanguageButton, LanguageSwapper}