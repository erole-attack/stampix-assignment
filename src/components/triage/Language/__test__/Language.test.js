import Language from '../Language'
import { LanguageContextProvider } from '../../../../context/context'
import React from "react"
import renderer from "react-test-renderer"

describe("Language", () => {
   it("renders correctly", () => {
      const tree = renderer
         .create(
            <LanguageContextProvider>
               <Language options={['NL', 'FR', 'EN']} />
            </LanguageContextProvider>
         )
         .toJSON()
      expect(tree).toMatchSnapshot()
   })
})