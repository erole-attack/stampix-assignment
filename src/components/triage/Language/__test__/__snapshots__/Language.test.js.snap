// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language renders correctly 1`] = `
<div
  className="sc-bdfBwQ iqcZPh"
>
  <button
    className="sc-gsTCUz bdboKt"
    onClick={[Function]}
    selected={false}
  >
    NL
  </button>
  <button
    className="sc-gsTCUz bdboKt"
    onClick={[Function]}
    selected={false}
  >
    FR
  </button>
  <button
    className="sc-gsTCUz nwHCe"
    onClick={[Function]}
    selected={true}
  >
    EN
  </button>
</div>
`;
